src/components/02_atoms/FileUpload/FileUpload.js

```diff
@@ -1,62 +1,155 @@
+`;
+
+const Error = styled('div')`
+  color: red;
+
+  > ul {
+    > li {
+      margin-bottom: 10px;
+    }
+
+    p {
+      margin: 0;
+    }
+  }
+`;
+
+class FileUpload extends Component {
   static propTypes = {
     entityTypeId: PropTypes.string.isRequired,
     bundle: PropTypes.string.isRequired,
     fieldName: PropTypes.string.isRequired,
     onFileUpload: PropTypes.func.isRequired,
+    inputProps: PropTypes.shape({
+      multiple: PropTypes.bool,
+      file_extensions: PropTypes.string,
+      max_filesize: PropTypes.number,
+    }).isRequired,
   };
 
+  /**
+   * Initial state
+   */
   state = {
-    drag: false,
-    file: null,
+    total: 0,
+    files: [],
+    errors: [],
+    filesLength: 0,
   };
 
+  /**
+   * Will set the border of the element to red and
+   * drop effect on drag over.
+   * @param {Event} event
+   */
   onDragOver = event => {
     event.stopPropagation();
     event.preventDefault();
-    this.setState({
-      drag: true,
-    });
+    this.setBorder('red');
     event.dataTransfer.dropEffect = 'dragend';
   };
 
+  /**
+   * Resets the border on drag leave.
+   * @param {Event} event
+   */
   onDragLeave = event => {
     event.stopPropagation();
     event.preventDefault();
-    this.setState({
-      drag: false,
-    });
+    this.setBorder();
   };
 
+  /**
+   * Will read the file/s that are dropped in the drop area.
+   * @param {Event} event
+   */
   onDrop = event => {
     event.stopPropagation();
     event.preventDefault();
+    this.readFile(event.dataTransfer.files);
+  };
 
-    const { files } = event.dataTransfer;
+  /**
+   * Will call the click event on the input to open the file explorer
+   * to allow file/s to be selected.
+   */
+  onClick = () => {
+    this.input.click();
+  };
+
+  /**
+   * Sets the border of the element.
+   * @param {String} [color="grey"] Color of the border.
+   */
+  setBorder = (color = 'grey') => {
+    const { element } = this;
+    if (element) {
+      element.style.border = `2px dashed ${color}`;
+    }
+  };
+
+  /**
+   * Will get the selected file/s from the file explorer.
+   * @param {Event} event
+   */
+  getFiles = event => {
+    this.readFile(event.target.files);
+  };
+
+  /**
+   * Uploads the file to the server, updates the total and files state.
+   * @param {Object} file File to be uploaded.
+   */
+  uploadFile = file => {
     const reader = new window.FileReader();
 
     reader.onloadend = async ({ target: { readyState, result } }) => {
       if (readyState === window.FileReader.DONE) {
+        const {
+          resetState,
+          props: { entityTypeId, bundle, fieldName, onFileUpload },
+        } = this;
         const { buffer } = new Uint8Array(result);
-
         const token = await api('csrf_token');
+        // Replace file name, some reason any space doesn't work
+        // TODO: Find a way to fix this without changing the name
+        const fileName = file.name.replace(/[,#!$^&*;{}=\-+`~()[\] ]/g, '_');
```

Comment:
Did you try using ```encodeURIComponent``` for the filename inside api.js? It looks like encoding the filename there could fix the problem you ran into

Reply:
This didn't work for me, maybe you could take a look?

Reply:
We could fix that later :)